usr.sbin/inetd/inetd.c

```diff
@@ -2748,8 +2737,10 @@ rl_process(struct servtab *sep, int ctrl)
 			return 0;
 			break;
 		}
-        	if (getnameinfo((struct sockaddr *)&addr, len, hbuf, 
-			sizeof(hbuf), NULL, 0, NI_NUMERICHOST)) {
+
+        	if (getnameinfo((struct sockaddr *)&addr, 
+		    ((struct sockaddr *)&addr)->sa_len, hbuf, 
```

Comment:
I guess on BSD sockaddr has a non-standard sa_len field (I had used something else previously but changed it back to what Alex and Gabe had). I think this should work with recvmsg, recvmsg doesn't set the namelen field in the msghdr.